subsys/debug/Kconfig

```diff
@@ -49,6 +49,7 @@ config TRACING
 config ASAN
 	bool "Build with address sanitizer"
 	depends on ARCH_POSIX
+	default y if TEST
```

Comment:
So far you were speaking about enabling ASAN for hte native_posix board, but this enables it for all POSIX Arch boards. If anything this should be only on NATIVE_POSIX (& 64) not for all

Reply:
> Do you see an issue to enable ASAN for all POSIX arch boards if they are running test?

Yep. Given that the nrf52_bsim is onyl relevant for BT developers, and has a very different aim than native_posix I would really prefer to not enable ASAN for it by default. So please do change this default to only apply to native_posix(64)